djvm/src/main/kotlin/net/corda/djvm/rewiring/SandboxClassLoader.kt

```diff
@@ -210,8 +215,8 @@ class SandboxClassLoader private constructor(
             logger.trace("Class {} already loaded", request.qualifiedClassName)
             return loadedClass
         } else if (analysisConfiguration.isPinnedClass(requestedPath)) {
-            logger.debug("Class {} is loaded unmodified", request.qualifiedClassName)
-            return loadUnmodifiedClass(requestedPath)
+            logger.error("Class {} should not be loaded here", request.qualifiedClassName)
+            throw SandboxClassLoadingException(context)
```

Comment:
Why it was necessary to change from returning `loadUnmodifiedClass` to throwing `SandboxClassLoadingException`?

Reply:
It turns out that `loadUnmodifiedClass()` is no longer called because loading "pinned" classes now happens naturally as part of ordinary class loading. So this exception is alerting us to something that shouldn't have happened.